java/team25core/ObjectDetectionTask.java

```diff
@@ -0,0 +1,250 @@
+    {
+        initVuforia(hardwareMap);
+
+        // if (ClassFactory.getInstance().canCreateTFObjectDetector()) {
+            initTfod(hardwareMap);
+//        } else {
+//            telemetry.addData("Sorry!", "This device is not compatible with TFOD");
+//        }
+    }
+
+    public void rateLimit(int ms)
+    {
+        this.rateLimitMs = ms;
+    }
+
+    public void setDetectionKind(DetectionKind detectionKind)
+    {
+        this.detectionKind = detectionKind;
+    }
+    //this will start tfod activation and start detecting
+    @Override
+    public void start()
+    {
+        tfod.activate();
+
+        if (rateLimitMs != 0) {
+            timer = new ElapsedTime(ElapsedTime.Resolution.MILLISECONDS);
+        }
+    }
+
+    @Override
+    public void stop()
+    {
+        tfod.deactivate();
+        robot.removeTask(this);
+    }
+
+    public static ObjectKind isObject(Recognition object)
+    {
+        if (object.getLabel().equals(LABEL_OBJECT1)) {
+            return ObjectKind.OBJECT1_KIND;
+        } else {
+            return ObjectKind.UNKNOWN_KIND;
+        }
+    }
+    //if recognize anything will add to que
+    protected void processEverything(List<Recognition> objects)
+    {
+        if (objects.size() > 0) {
+            robot.queueEvent(new ObjectDetectionEvent(this, EventKind.OBJECTS_DETECTED, objects));
+        }
+    }
+    //only adds objects which will make event and add to que
+    protected void processObject1(List<Recognition> objects)
+    {
+        List<Recognition> objectList = new ArrayList<>();
+        for (Recognition object : objects) {
+            if (isObject(object) == ObjectKind.OBJECT1_KIND){
+                objectList.add(object);
+            }
+        }
+
+        if (!objectList.isEmpty()) {
+            robot.queueEvent(new ObjectDetectionEvent(this, EventKind.OBJECTS_DETECTED, objectList));
+        }
+    }
+
+//    protected void processObject2(List<Recognition> objects)
+//    {
+//        List<Recognition> objectList = new ArrayList<>();
+//        for (Recognition object : objects) {
+//            if (isObject(object) == ObjectKind.OBJECT2_KIND) {
+//                objectList.add(object);
+//            }
+//        }
+//
+//        if (!objectList.isEmpty()) {
+//            robot.queueEvent(new ObjectDetectionEvent(this, EventKind.OBJECTS_DETECTED, objectList));
+//        }
+//    }
+
+    //timeslice calls to get information from recognition
+    protected void processDetectedObjects(List<Recognition> objects)
+    {
+        if (objects == null || objects.isEmpty()) {
+            return;
+        }
+
+        switch (detectionKind) {
+            case EVERYTHING:
+                processEverything(objects);
+                break;
+//            case OBJECT2_DETECTED:
+//                processObject2(objects);
+//                break;
+            case OBJECT1_DETECTED:
+                processObject1(objects);
+                break;
```

Comment:
This is where you'd do

```
processObject(objects, ObjectKind.OBJECT1_KIND)
```

or OBJECT2_KIND depending upon which switch case detectionKind is.